### PR TITLE
The Start Now verb now has undo options

### PR DESCRIPTION
**Pull request by kriskog**
## About The Pull Request
Psssst! I am being preemptive about this cause I WILL fuck up eventually and scary buttons should have an undo! 

The "Start Now" verb will now ask you for confirmation if you are not connecting from localhost. It is also possible to cancel immediate start by using the verb again, or using the delay pre-game verb.

## Why It's Good For The Game
Coder man habits on live servers bad.

## Changelog
:cl: Skoglol
admin: The "Start Now" verb can now be cancelled by using it again or setting a custom pre-game delay.
/:cl:

### CODE REVIEW

**Comment by ninjanomnom** [comment]
Put a link in a message sent to admins that allows them to prevent the early start.

Also I would prefer to not have to deal with this most of the time in local dev, make it so when someone connected locally does this command there is no delay.

**Comment by kriskog** [comment]
Hm links can be cancelled by all the other admins so yeah I guess thats good. Upping to 10 seconds and making instant for localhost.

**Comment by PKPenguin321** [comment]
I'd prefer if it still started ASAP but the verb had a Yes/No confirmation prompt (can default to No if necessary)

**Comment by Grazyn** [comment]
Yeah I don't see the reason for the delay if it's instant for localhost, when a big prompt with "WARNING YOU'RE ON A LIVE SERVER WITH REAL PEOPLE IN IT!!!" (that doesn't pop-up for localhost) should be enough to break any bad habit. Yes admins wouldn't be able to stop it, but it means the coder was already malicious so he could've fucked up the server in any other way.

**Comment by kriskog** [comment]
@PKPenguin321 How often do you purposefully use this on live servers?

Prompting when non-localhost might be fine though, I will leave the cancels in though if used before initialization is done.

**Comment by kriskog** [comment]
Non localhosts will now get a confirmation dialog box when they use the start now verb, and it is once again instant. You can still cancel by using the verb again or delaying pre-game.

File: code/modules/admin/admin.dm [hunk 1]
```diff
@@ -561,20 +561,29 @@
...
-			msg = " (The server is still setting up, but the round will be \
-				started as soon as possible.)"
-		message_admins("<font color='blue'>\
-			[usr.key] has started the game.[msg]</font>")
-		SSblackbox.record_feedback("tally", "admin_verb", 1, "Start Now") //If you are copy-pasting this, ensure the 2nd parameter is unique to the new proc!
-		return 1
+		if(!SSticker.start_immediately)
+			var/localhost_addresses = list("127.0.0.1", "::1")
+			if(!(isnull(usr.client.address) || (usr.client.address in localhost_addresses)))
+				if(alert("Are you sure you want to start the round?","Start Now","Start Now","Cancel") == "Cancel")
```
Comment (ninjanomnom) [inline code/modules/admin/admin.dm]:
```suggestion
				if(alert("Are you sure you want to start the round?","Start Now","Start Now","Cancel") != "Start Now")
```
This avoids issues like if the window is closed without either of the buttons being pressed

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

File: code/modules/admin/admin.dm (hunk 1 above)
Comment (kriskog) [reply code/modules/admin/admin.dm]: Thanks

**Review by ninjanomnom — APPROVED** [review]

**Pull request merged**